### ISSUE TITLE
TagPicker: few invalid aria attributes

### ISSUE THREAD

**Issue by prmittal** [OfficeDev/office-ui-fabric-react#3734]
#### Describe the issue:

Example taken from https://developer.microsoft.com/en-us/fabric#/components/pickers

![image](https://user-images.githubusercontent.com/11937034/35038466-e68a3f88-fba0-11e7-997d-d5c6ebb55702.png)
(Above image is taken after running Keros fast pass)

#### Expected behavior:

All aria attributes should be valid

**Comment by battletoilet** [comment]
Hey @joschect, can you see which ARIA tag is invalid here and do we have a bug. Also @cschlechty open for comments as well.

**Comment by joschect** [comment]
@battletoilet  and @cschlechty  I'm pretty sure both of those are valid aria attributes. Please correct me if I'm wrong Chris. Also they should both be fine. Maybe it's upset with `sug--1` ?

**Comment by joschect** [comment]
@battletoilet  and @cschlechty I see the problem sug--1 doesn't exist, for some reason an extra - is being added. Should be a quick fix.

**Comment by joschect** [comment]
Now I'm confused again, it appears correct when I'm testing 
![image](https://user-images.githubusercontent.com/20801176/35119339-752da36c-fc49-11e7-9e96-b1ba41d07762.png)

**Comment by joschect** [comment]
@prmittal is it possible that you were looking at an old version?

**Comment by prmittal** [comment]
I tried the scenario on https://developer.microsoft.com/en-us/fabric#/components/pickers
and this is what I see:
![image](https://user-images.githubusercontent.com/11937034/35135509-b9465642-fd03-11e7-9a6f-e63875b80263.png)

**Comment by prmittal** [comment]
Sorry, closed the issue by mistake. 
For the version, it could be the link above has not been updated.

**Comment by aditima** [comment]
@joschect - the sug--1 seems to be coming from this code in BasePicker.tsx:

aria-activedescendant={ 'sug-' + this.suggestionStore.currentIndex }

From surrounding code, looks like currentIndex can possibly be -1.

If this is the invalid syntax, should be easily fixable in code.

**Comment by aditima** [comment]
sorry did not mean to close this.

**Comment by joschect** [comment]
Ah, that makes sense. That's an easy fix. I'll just have it be undefined if currentIndex < 0

**Comment by random-creator** [comment]
@joschect Is there a fix for this yet? I am seeing the same issue.

**Comment by Shubhangi-Priya** [comment]
@joschect Is there a fix for this yet? I am seeing the same issue.

**Comment by joschect** [comment]
@Shubhangi-Priya  I thought I had put a fix in but I clearly did not. Here's one now that will fix it #4238

**Comment by Shubhangi-Priya** [comment]
Thanks so much @joschect !

**Comment by joschect** [comment]
This will now be fixed in the next update of fabric!
